docs/README.md

```diff
@@ -168,4 +164,83 @@ Updated default.config NuGet package versions...
         VS.Redist.Common.AspNetCore.SharedFramework.x64.3.1, version: 3.1.2-servicing.20068.1
         VS.Redist.Common.AspNetCore.SharedFramework.x86.3.1, version: 3.1.2-servicing.20068.1
         ...
-</pre>
\ No newline at end of file
+</pre>
+
+## <a href="#codebase">Codebase</a>
+This section covers highlights of both the API and console codebases, respectively. 
+- The code base is hosted in the [insertions-client](https://github.com/dotnet/insertions-client) GitHub repository. 
+- The InsertionClient solution consists of docs, src, and tests
+
+
+### <a href="#corelib">Microsoft.DotNet.InsertionsClient.Core</a>
+The library publicly exposes the _IInsertionApi_ interface, enabling callers 
+to trigger the updating of SDK insertion file assets. 
+See _[How it works](#howitworks)_ for more deails.  In addition, the library 
+contains the models and untilities to support the version updates.
+
+#### IInsertionApi Interface 
+Defines how to interact with the library to update versions of targeted assets, such as _default.conf_.
+```csharp
+UpdateResults UpdateVersions(
+            IEnumerable<string> manifestFiles,
+            string defaultConfigFile,
+            IEnumerable<Regex> whitelistedPackages,
+            ImmutableHashSet<string>? packagesToIgnore,
+            string? accessToken,
+            string? propsFilesRootDirectory,
+            Predicate<Build>? buildFilter);
+```
+- __manifestFiles:__ The paths to all the manifest.json files to be inserted.
+- __defaultConfigFile:__ Full path to _default.config_ file.
+- __whitelistedPackages:__ Regex patterns matching with the packages that are allowed to be updated. If the set is empty,
+ any package is allowed be updated unless specified in packagesToIgnore.
+- __packagesToIgnore:__ HashSet{string} of packages to ignore.
+- __accessToken:__ Access token used when connecting to nuget feed.
+- __propsFilesRootDirectory:__ Directory that will be searched for props files.
+- __buildFilter:__ Predicate to determine if a build should be included in the insertion process.
+- Returns an UpdateResults instance, detailing the operation's outcome.
+
+
+#### IInsertionApiFactory Interface 
+Factory of instances, containing a single method.
+```csharp
+IInsertionApi Create(TimeSpan? maxWaitSeconds = null, TimeSpan? maxDownloadSeconds = null, int? maxConcurrency = null);
+```
+
+#### <a href="corelibexamples"/>Examples</a>
+##### Creating IInsertionApi instance
+```csharp
+ IInsertionApiFactory apiFactory = new InsertionApiFactory();
+ IInsertionApi api = apiFactory.Create(MaxWaitDuration, MaxDownloadDuration, MaxConcurrency);
+```
+
+### <a href="#console">Microsoft.DotNet.InsertionsClient.Console</a>

```

Comment:
Just an idea: should we include some examples on how to use console?

Reply:
Console examples are already available under "Usage" header. Should we add more?